bin/zbm_kcl

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+zbm_load_lib() {
+  lib="${1?no library specified for load}"
+
+  : "${ZBM_MODULEDIR:=/usr/lib/dracut/modules.d/90zfsbootmenu}"
+
+  if [ -r "${ZBM_MODULEDIR}/lib/${lib}" ]; then
+    # shellcheck disable=SC1090
+    source "${ZBM_MODULEDIR}/lib/${lib}"
+  else
+    echo "ERROR: ${ZBM_MODULEDIR}/lib/${lib} not found" >&2
+    echo "Set ZBM_MODULEDIR to the root of the ZBM dracut module" >&2
+    exit 1
+  fi
+}
+
+usage() {
+  cat <<-EOF
+	USAGE: $0 [-a <arg>] [-r <arg>] [-e] [-d] [-v] [bootenv]
+        Review or update kernel command line (KCL) associated with <bootenv>
+        When the boot environment is specified, the current root will be used
+
+	-a <arg>: Append an argument
+	-r <arg>: Remove an argument
+	-e: Open the KCL for editing in \$EDITOR
+	-d: Remove entire command line
+	-v: Enable verbose output
+	EOF
+}
+
+strip_kcl() {
+  awk '
+  BEGIN { first = 1; }
+
+  /^$/ { exit; }
+
+  {
+    gsub("[[:space:]]*#.*", "");
+
+    if (length == 0) next;
+
+    if (first == 1) {
+      first = 0;
+    } else {
+      printf " ";
+    }
+
+    printf "%s", $0;
+  }
+  '
+}
+
+delete() {
+  local bootenv="${1}"
+
+  if ! zfs list -o name -H "${bootenv}" >/dev/null 2>&1; then
+    zerror "no boot environment specified"
+    usage
+    return 1
+  fi
+
+  if ! zfs inherit org.zfsbootmenu:commandline "${bootenv}"; then
```

Comment:
How should things be handled in the event the property is set for `ztest/ROOT` ? delete() in this case will just restore the parent property to the BE.

Reply:
That's the desired behavior. It's nice to be able to customize a specific boot environment and just use `-d` to clear the customization and restore the inherited behavior. I also think trying to be a bit craftier is likely to blow up strangely in some use cases.

Reply:
I see, yup! Good call. I missed that you can specify a boot environment instead of just relying on finding the ZFS filesystem mounted at `/`.